docs/development/pigs/pig-030.rst

```diff
@@ -0,0 +1,447 @@
+
+    model = crab_point_source_model()
+
+    model_npred = NPredSystematicsModel.from_dataset(
+        model=model,
+        dataset=dataset,
+        exposure_modifier=exposure_modifier,
+    )
+    dataset.models = model_npred
+
+
+2. PSF scaling by radial axis
+
+.. code-block:: python
+
+    psf_parameter = Parameter('psf_scaling', 1)
+    psf_parameter.prior = GaussianPrior(mu=1, sigma=0.2)
+
+    scale_effect = ScalingEffect()
+
+    psf_modifier = Modifier(psf_parameter, scale_effect, axis='rad') # applied to axis of IRF
+
+    model = crab_point_source_model()
+
+    model_npred = NPredSystematicsModel.from_dataset(
+        model=model,
+        dataset=dataset,
+        psf_modifier=psf_modifier,
+    )
+    dataset.models = model_npred
+
+3. Energy scale bias in EDisp
+
+.. code-block:: python
+
+    edisp_parameter = Parameter('edisp_scaling', 0)
+    edisp_parameter.prior = GaussianPrior(mu=0, sigma=0.2)
+    shift_effect = LogShiftEffect()
+
+    edisp_modifier = Modifier(edisp_parameter, shift_effect, axis='energy')
+
+    model = crab_point_source_model()
+
+    model_npred = NPredSystematicsModel.from_dataset(
+        model=model,
+        dataset=dataset,
+        edisp_modifier=edisp_modifier,
+    )
+    dataset.models = model_npred
+
+4. All of the above at once
+
+.. code-block:: python
+
+
+    model_npred = NPredSystematicsModel.from_dataset(
+        model=model,
+        dataset=dataset,
+        exposure_modifier=exposure_modifier,
+        psf_modifier=psf_modifier,
+        edisp_modifier=edisp_modifier,
+    )
+    dataset.models = model_npred
+
+5. PSF modification with increasing Energy
+
+.. code-block:: python
+
+    # apply scaling linear as scaling = slope * energy + intersect
+    psf_parameter_slope = Parameter('psf_scaling_slope', 1)
+    psf_parameter_slope.prior = GaussianPrior(mu=1, sigma=0.2)
+    psf_parameter_intersect = Parameter('psf_scaling_intersect', 1)
+    psf_parameter_intersect.prior = GaussianPrior(mu=1, sigma=0.2)
+
+    offset_effect = LinearScalingEffect(depends_on_axes=['energy_true']) # change scaling with energy_true
+
+    psf_modifier = Modifier([psf_parameter_slope, psf_parameter_intersect], offset_effect, axis='rad') # applied to rad axis
+
+    model = crab_point_source_model()
+
+    model_npred = NPredSystematicsModel.from_dataset(
+        model=model,
+        dataset=dataset,
+        psf_modifier=psf_modifier,
+    )
+    dataset.models = model_npred
+
+
+Implementation
+==============
+
+Modifier
+--------
+
+A ``Modifier`` links a single instance or list of ``Parameter`` instances to an ``Effect``.
+The modifier is responsible for calling the effect with the current parameter values and applying it to
+either the IRF *data* or its *axis* coordinates (e.g. shifting the ``energy`` axis).
+In the latter case the transformation defined by the ``Effect`` is applied by interpolating the ``IRF`` to new transformed coordinates.
+
+.. code-block:: python
+
+    class Modifier:
```

Comment:
Shouldn't it be a `Model`? 
In that case, internally, it is a model that take a `MapCoord` and yields a `MapCoord` i.e. from R^3 to R^3.

You could then represent it as `IRFmodel(irf_map, spectral_modifier, spatial_modifier, coord_effect)`
Then when evaluating the IRf model you have:
- new_coords = coord_effect(coords)
- return irf_map(new_coords)*spectral_modifier(new_coords[energy])*spatial_modifier(new_coords[space])